mapie/control_risk/ltt.py

```diff
@@ -0,0 +1,134 @@
+import warnings
+
+from typing import Tuple, List, Optional, Any
+from mapie._typing import NDArray, ArrayLike
+
+import numpy as np
+
+from .p_values import compute_hoeffdding_bentkus_p_value
+
+
+def _ltt_procedure(
+    r_hat: NDArray,
+    alpha_np: NDArray,
+    delta: Optional[float],
+    n_obs: int
+) -> Tuple[List[List[Any]], NDArray]:
+    """
+    Apply the Learn-Then-Test procedure for risk control.
+    This procedure is called in ``MapieMultiLabelClassifier``
+    if ``metric=precision``.
+    This will apply learn then test procedure for
+    precision control.
+    Note that we will do a multiple test for ``r_hat`` that are
+    less than level ``alpha_np``.
+
+    The procedure follows the instructions in [1]:
+        - Calculate p-values for each lambdas descretized
+        - Apply a fwer algorithm, here Bonferonni correction
+        - Return the index lambdas that give you the control
+        at alpha level
+
+    Parameters
+    ----------
+    r_hat: NDArray of shape (n_lambdas, ).
+        Empirical risk with respect
+        to the lambdas.
+        Here lambdas are thresholds that impact decision making,
+        therefore empirical risk.
+
+    alpha_np: NDArray of shape (n_alpha, ).
+        Contains the different alphas control level.
+        The empirical risk should be less than alpha with
+        probability 1-delta.
+
+    delta: float.
+        Probability of not controlling empirical risk.
+        Correspond to proportion of failure we don't
+        want to exceed.
+
+    Returns
+    -------
+    valid_index: List[List[Any]].
+        Contain the valid index that satisfy fwer control
+        for each alpha (length aren't the same for each alpha).
+
+    p_values: NDArray of shape (n_lambda, n_alpha).
+        Contains the values of p_value for different alpha.
+
+    References
+    ----------
+    [1] Angelopoulos, A. N., Bates, S., Candès, E. J., Jordan,
+    M. I., & Lei, L. (2021). Learn then test:
+    "Calibrating predictive algorithms to achieve risk control".
+    """
+    if delta is None:
```

Comment:
check already done in MapieMultilabelClassifier and this function can not be specific to precision control

Reply:
As delta is Optional[float], not testing delta will result in a make type-check error as None cannot be compute with float. So even if the test seems to be useless, it is to me better to keep this test.
Do you still think that I should do otherwise?